projects/lib/src/chessgame.h

```diff
@@ -137,6 +142,7 @@ class LIB_EXPORT ChessGame : public QObject
 		QSemaphore m_pauseSem;
 		QSemaphore m_resumeSem;
 		GameAdjudicator m_adjudicator;
+		const Tournament* m_tournament;
```

Comment:
I can see the logic there. Anyway, I thought I'd put code behind my vision so I hope you don't mind that I rewrote most of your code. It's still not as pretty as I would like but achieves a couple of goals:

* Avoid copy-pasting the same board flipping logic to multiple places
* Make sure `ChessGame` is not dependent on `Tournament`
* Put `GauntletTournament` specific code inside the `GauntletTournament` class, which is cleaner than checking the tournament type from outside.